### ISSUE TITLE
Fix coercion of Strings to Lists in MergeFlags

### ISSUE THREAD

**Issue by bdbaddog** [SCons/scons#2961]
This issue was originally created at: 2014-07-13 05:15:21.
This issue was reported by: `dirkbaechle`.
dirkbaechle said at 2014-07-13 05:15:21
>As reported on the mailing list: https://scons.tigris.org/ds/viewMessage.do?dsForumId=1272&viewType=browseAll&dsMessageId=2876001 the `MergeFlags()` method doesn't handle appending flags to already existing values very well. The coercion of types in `ParseFlags()` (as copied from the `Append` method) should get amended to properly handle the following use case:
```py
env.MergeFlags({'CCFLAGS' : '-whatever -O3'})
```
where `CCFLAGS` is initially empty.

dirkbaechle said at 2014-07-13 05:33:10
>added dependency

dirkbaechle said at 2014-07-13 07:46:05
>*** Issue 2402 has been marked as a duplicate of this issue. ***

dirkbaechle said this issue depends on #2962 at 2014-07-13 05:33:10.

dirkbaechle said this issue is duplicated by #2402 at 2014-07-13 07:46:05.

**Comment by mwichmann** [comment]
Trying to figure this out: is this actually expected to work:
```
env.MergeFlags({'CCFLAGS' : '-whatever -O3'})
```
that is, passing a string containing multiple args as opposed to a list of args?
The comment about this being from `ParseFlags` seems to be a red herring - since a dictionary is passed, `ParseFlags` is not called:
```
        if not SCons.Util.is_Dict(args):
            args = self.ParseFlags(args)
```
